src/services/study.ts

```diff
@@ -46,3 +55,56 @@ export const getEmissionSourceStatus = (
 
   return EmissionSourcesStatus.Waiting
 }
+
+export const downloadStudySubPosts = async (
+  study: FullStudy,
+  post: string,
+  subPost: SubPost,
+  emissionSources: FullStudy['emissionSources'],
+  emissionFactors: EmissionFactorWithMetaData[],
+  t: ReturnType<typeof useTranslations>,
+  tQuality: ReturnType<typeof useTranslations>,
+) => {
+  const columns = [
+    'validation',
+    'sourceName',
+    'sourceCharacterization',
+    'sourceValue',
+    'sourceComment',
+    'sourceQuality',
+    'emissionName',
+    'emissionValue',
+    'emissionSource',
+    'emissionQuality',
+  ]
+    .map((key) => t(key))
+    .join(';')
+
+  const csvRows = emissionSources.map((emissionSource) => {
+    const emissionFactor = emissionFactors.find((factor) => factor.id === emissionSource.emissionFactor?.id)
+    return [
+      emissionSource.validated ? t('yes') : t('no'),
+      emissionSource.name || '',
+      emissionSource.caracterisation || '',
+      emissionSource.value || '0',
+      emissionSource.comment || '',
+      getQuality(getQualityRating(emissionSource), tQuality),
+      emissionFactor?.metaData?.title || t('noFactor'),
+      emissionFactor?.totalCo2 || '',
+      emissionFactor?.source || '',
+      emissionFactor ? getQuality(getQualityRating(emissionFactor), tQuality) : '',
+    ].join(';')
+  })
+
+  const totalEmissions = emissionSources.reduce((sum, item) => sum + (item.value || 0), 0)
+  const totalRow = [t('total'), '', '', totalEmissions].join(';')
+
+  // TO DO : Ajouter la ligne des incertitudes
```

Comment:
```suggestion
  // TODO : Ajouter la ligne des incertitudes
```
c'est pas vraiment un nitpick, toujours bien de faire un ctrl f todo pour voir ceux la